driver-core/src/main/com/mongodb/client/model/expressions/StringExpression.java

```diff
@@ -35,13 +37,22 @@ public interface StringExpression extends Expression {
 
     StringExpression substr(IntegerExpression start, IntegerExpression length);
 
-    default StringExpression substr(int start, int length) {
+    default StringExpression substr(final int start, final int length) {
         return this.substr(of(start), of(length));
     }
 
     StringExpression substrBytes(IntegerExpression start, IntegerExpression length);
 
-    default StringExpression substrBytes(int start, int length) {
+    default StringExpression substrBytes(final int start, final int length) {
         return this.substrBytes(of(start), of(length));
     }
+
+    IntegerExpression parseInteger();
+
+    DateExpression parseDate();
+
+    DateExpression parseDate(StringExpression format, StringExpression timezone);
+
+    @Beta(Beta.Reason.CLIENT)
```

Comment:
Discussed here: https://github.com/mongodb/mongo-java-driver/pull/1050#discussion_r1030873542

Reply:
But it's still not clear why this one method would remain as Beta if Beta were to be removed from the rest of the API.

Reply:
The idea was that there was a sensitive decision with respect to that method that needed to be explicitly decided, and it seems that we have no process that ensures the issue would be addressed at the time that we removed @Beta from the API generally.

But it's a moot point because the method was removed from the API.